### PR TITLE
[13.0][FIX] stock_picking_report_valued: Refactor code to avoid decimal differences

### PR DESCRIPTION
**Pull request by carlosdauden**
Related PRs:
https://github.com/OCA/stock-logistics-reporting/pull/130
https://github.com/OCA/stock-logistics-reporting/pull/131

@Tecnativa 
TT17492 TT26034 TT33957 TT34954

### CODE REVIEW

**Comment by carlosdauden** [comment]
@emagdalenaC2i can you check this?

File: stock_picking_report_valued/models/stock_move_line.py [hunk 1]
```diff
@@ -42,38 +43,42 @@ class StockMoveLine(models.Model):
...
-            )
-            if sale_line.company_id.tax_calculation_rounding_method == (
-                "round_globally"
+            quantity = line.qty_done or line.product_qty
+            # If order line quantity match with move line quantity get from original SOL
+            if (
+                float_compare(
+                    quantity,
+                    line.sale_line.product_uom_qty,
+                    precision_digits=price_digits,
```
Comment (carlosdauden) [inline stock_picking_report_valued/models/stock_move_line.py:66]:
```suggestion
                    precision_digits=line.product_uom.digits,
```

**Comment by carlosdauden** [comment]
ping @chienandalu @sergio-teruel @emagdalenaC2i

**Review by rousseldenis — APPROVED** [review]
Code review.

**Review by sergio-teruel — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Cherry-pick for v14 at 343503b0361b47fc56a239314445ef442b376f20

**Pull request merged**